sheetstorm/management/commands/data_load.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from users.factories import UserFactory
+
+User = get_user_model()
+
+user_type_options = {"admin": "admin", "manager": "manager", "employee": "employee"}
+
+
+class Command(BaseCommand):
+    help = "Create initial sample data for SheetStorm application testing."
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            "-a", "--admin", type=int, help="Indicates the maximum number of admins to be in the database"
+        )
+        parser.add_argument(
+            "-e", "--employee", type=int, help="Indicates the maximum number of employees to be in the database"
+        )
+        parser.add_argument(
+            "-m", "--manager", type=int, help="Indicates the maximum number of managers to be in the database"
+        )
+
+    @staticmethod
+    def add_user(user_type, number_of_users_to_create):
+        if user_type == user_type_options["admin"]:
+            factory_parameters = {"user_type": User.UserType.ADMIN.name, "is_staff": True}
+            password = "admin1234"
+
+        if user_type == user_type_options["employee"]:
+            factory_parameters = {"user_type": User.UserType.EMPLOYEE.name}
+            password = "passwduser"
+
+        if user_type == user_type_options["manager"]:
+            factory_parameters = {"user_type": User.UserType.MANAGER.name}
+            password = "passwduser"
+
+        for new_user in range(number_of_users_to_create):
+            user = User.objects.filter(email=f"user.{user_type}{new_user}@codepoets.it").first()
+            if user is None:
+                user = UserFactory(**factory_parameters, email=f"user.{user_type}{new_user}@codepoets.it")
+                user.set_password(password)
```

Comment:
I can see in PyCharm warnings about **factory_parameters** and **password**: _local variable might be referenced before assignment_.

This is because those variables are always defined **under a condition** and although the code is working fine (since exactly one condition is always fulfilled) we should treat it as an error. You can solve it in two ways:

1. Assign those variables before the first if statement. Those can be values for one of the user types (like the employee). Then replace those values under **if** statements accordingly.

pseudocodishly:
```
factory_parameters = {"user_type": EMPLOYEE}
password = "passwduser"

if user_type is a manager:
  factory_parameters["user_type"] =  EMPLOYEE
elif user_type is an admin:
  factory_parameters["user_type"] =  ADMIN
  factory_parameters["is_staff"] =  True

```

2. Turn those three _if_ statements into **a single consecutive statement**: **if** -> **elif** -> **else**. The **else** at the end guarantees that the following section of code will always be executed and therefore variables are guaranteed to be assigned.

pseudocodishly:
```
if user_type is an admin:
  factory_parameters = {"user_type": ADMIN, "is_staff": True}
  password = "admin1234"
elif user_type is an employee:
  factory_parameters = {"user_type": EMPLOYEE}
  password = "passwduser"
else:
  factory_parameters = {"user_type": User.UserType.MANAGER.name}
  password = "passwduser"

```

Reply:
Done. I declare a variable _password_for_every_user_  now and also I turned _if_ statements into a _**if**_ -> _**elif**_ -> _**else**_ , so you shouldn't get this warnings again.